Source/engine/render/scrollrt.cpp

```diff
@@ -747,13 +738,62 @@ void DrawDungeon(const Surface &out, Point tilePosition, Point targetBufferPosit
 	if (TileContainsDeadPlayer(tilePosition)) {
 		DrawDeadPlayer(out, tilePosition, targetBufferPosition);
 	}
-	int8_t playerId = dPlayer[tilePosition.x][tilePosition.y];
-	if (static_cast<size_t>(playerId - 1) < Players.size()) {
-		DrawPlayerHelper(out, Players[playerId - 1], tilePosition, targetBufferPosition);
+	Player *player = PlayerAtPosition(tilePosition);
+	if (player != nullptr) {
+		// If sprite is moving southwards, we want to draw it offset from the tile it's moving to, so we need negative ID
+		int8_t playerId = (player->_pmode != PM_WALK_SOUTHWARDS) ? player->getId() + 1 : -player->getId() - 1;
+
+		if (dPlayer[tilePosition.x][tilePosition.y] == playerId) {
+			auto tempTilePosition = tilePosition;
+			auto tempTargetBufferPosition = targetBufferPosition;
+
+			// Offset the sprite to the tile it's moving from
+			if (player->_pmode == PM_WALK_SOUTHWARDS) {
+				switch (player->_pdir) {
+				case Direction::SouthWest:
+					tempTargetBufferPosition += { TILE_WIDTH / 2, -TILE_HEIGHT / 2 };
+					break;
+				case Direction::South:
+					tempTargetBufferPosition += { 0, -TILE_HEIGHT };
+					break;
+				case Direction::SouthEast:
+					tempTargetBufferPosition += { -TILE_WIDTH / 2, -TILE_HEIGHT / 2 };
+					break;
+				}
+				tempTilePosition += Opposite(player->_pdir);
+			}
+			DrawPlayerHelper(out, *player, tempTilePosition, tempTargetBufferPosition); // BUGFIX: Player sprite gets cut off walking in front of certain corners in caves

```

Comment:
Is this new since all the s-walk changes, or how it was originally?

Reply:
Sideways walking also had its own unique handling for tile occupation, which was removed along with the southwards movement logic.

Reply:
The `WalkParameters` `startOffset` for `West` was the same as `SouthWest`, and `East` was the same was `SouthEast`.

In `getOffsetForWalking()`, it appears that moving `West` would shift the player to the right by `TILE_WIDTH` and moving `East` would shift the player to the left by `TILE_WIDTH`.

Reply:
Walking sideways also applied a manually adjustment of 1 tile on the positive y axis for `West` and 1 tile on the positive x axis for `East`. I'm really having a hard time making sense how it all worked, but I'm guessing it was just a hack for basically doing the same thing south walking was doing to a degree, where the player was being drawn as a part of a later drawn tile to get draw priority.